### PR TITLE
WIP async utilities

### PR DESCRIPTION
**Pull request by sadasant**
Utilities for async/await or promises in general! What do you guys think?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,3 +193,14 @@ Maps a function over a recursive iterable. Works by default for nested Arrays, n
...
+## Async functions
+
+## asyncMap
+Like lodash/fp map, but accepts promises. Promises run in parallel.
+
+## asyncReduce
+Like lodash/fp reduce, but accepts promises. Promises run in sequence.
+
+## asyncIterator
+Reduces over a promise based iterator.
```
Comment (sadasant) [inline README.md]: This currently doesn't reduce that, it just groups the results into an array.

File: package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
   "author": "Samuel Greene <sgreene@smartprocure.us>",
   "license": "MIT",
   "dependencies": {
+    "bluebird": "^3.5.0",
```
Comment (daedalus28) [inline package.json]: we should not take bluebird as a dependency for futil - those methods like async map/reduce/etc we should split out to a separate `bluebird-fp` library

File: README.md [hunk 3]
```diff
@@ -372,6 +372,29 @@ Opposite of `isBlank`
...
+## asyncMap
+Like lodash/fp map, but accepts promises. Promises run unordered.
+
+## asyncSeries
+Like lodash/fp map, but accepts promises. Promises run in order.
+
+## asyncReduce
+Like lodash/fp reduce, but accepts promises. Promises run in order.
+
+## asyncProps
```
Comment (daedalus28) [inline README.md]: Async Props isn't int here, but it's also quite literally Promise.props from bluebird

File: src/async.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import _ from 'lodash/fp'
+import Promise from 'bluebird'
+
+export let asyncMap = _.curry((fn, obj) => Promise.map(obj, fn))
```
Comment (dshishkov) [inline src/async.js]: If we are just doing a curry here and flipping the param order of the bluebird method, how can we preserve the options param that Bluebird.map can take, to control concurrency for example. maybe have asyncMapOpt that does _.curry(opts, fn, obj) and your asyncMap becomes asyncMapOpt({}) or something like that

Comment (daedalus28) [reply src/async.js]: exactly why this really belongs as a bluebird-fp lib - we're really just currying and re-arging the bluebird stuff here

Comment (sadasant) [reply src/async.js]: But so this `bluebird-fp` should have `asyncFlow`? isn't it better to call this separate library something like `async-fp`??

Comment (daedalus28) [reply src/async.js]: just the ones that are just rearging/currying bluebird methods

**Comment by daedalus28** [comment]
Closing in favor of #194

**Pull request closed without merging**